src/stores/blockstore.js

```diff
@@ -52,19 +52,22 @@ exports.setUp = (basePath, BlobStore, locks) => {
       }
 
       const p = multihashToPath(key, extension)
+      const deferred = defer()
+
+      lock(p, (release) => {
+        const ext = extension === 'data' ? 'protobuf' : extension
+        pull(
+          store.read(p),
+          pull.collect(release((err, data) => {
+            if (err) return deferred.abort(err)
+            deferred.resolve(pull.values([
+              new Block(Buffer.concat(data), ext)
+            ]))
+          }))
+        )
+      })
```

Comment:
update: nvm. It is because of `interface-pull-blob-store` only streams interface. We should consider to have a sync interface too on the blob-store.